File: ios/ReactNativeRust/RNChannel.m
//
//  RNChannel.m
//  ReactNativeRust
//
//  Created by Phạm Tiến on 19/04/2023.
//

#import <Foundation/Foundation.h>
#import "RNChannel.h"

@interface RCT_EXTERN_MODULE(RNChannel, RCTEventEmitter)
RCT_EXTERN_METHOD(response:(NSString *)uuid
                  response:(NSString *)response
                  resolver:(RCTPromiseResolveBlock)resolver
                  rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(call:(NSString *)uuid
                  method:(NSString *)method
                  request:(NSString *)request
                  resolver:(RCTPromiseResolveBlock)resolver
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/ReactNativeRust/RNChannel.swift
//
//  RNChannel.swift
//  ReactNativeRust
//
//  Created by Phạm Tiến on 19/04/2023.
//

import Foundation
import Connect
import OSLog

@objc(RNChannel)
class RNChannel: RCTEventEmitter {
    override func supportedEvents() -> [String]! {
      return []
    }
    
    @objc
    func response(_ uuid: String, response: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        
    }
    
    @objc
    func call(_ uuid: String, method: String, request: String, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
        os_log("call success");
    }
}

